packages/polaris-middlewares/src/dates-filter/dates-filter-middleware.ts

```diff
@@ -0,0 +1,77 @@
+import {PolarisConnectionManager} from "@enigmatis/polaris-typeorm";
+import {EntityFilter, PolarisGraphQLContext, RealitiesHolder} from "@enigmatis/polaris-common";
+import {PolarisGraphQLLogger} from "@enigmatis/polaris-graphql-logger";
+import {entityFilterInputTypeName} from "@enigmatis/polaris-schema";
+
+export class DatesFilterMiddleware {
+    public readonly connectionManager?: PolarisConnectionManager;
+    public readonly realitiesHolder: RealitiesHolder;
+    public readonly logger: PolarisGraphQLLogger;
+
+    constructor(
+        logger: PolarisGraphQLLogger,
+        realitiesHolder: RealitiesHolder,
+        connectionManager?: PolarisConnectionManager,
+    ) {
+        this.connectionManager = connectionManager;
+        this.realitiesHolder = realitiesHolder;
+        this.logger = logger;
+    }
+
+    public getMiddleware() {
+        return async (
+            resolve: any,
+            root: any,
+            args: any,
+            context: PolarisGraphQLContext,
+            info: any,
+        ) => {
+            this.logger.debug('Data version middleware started job', context);
+            if(Object.keys(info.schema._queryType._fields).includes(info.fieldName)) {
```

Comment:
info.schema._queryType._fields
appears in every query? can be undefined?

Reply:
i've changed the code and now it is more clearer and understandable